subsys/net/lib/openthread/platform/radio.c

```diff
@@ -279,7 +274,8 @@ static void openthread_handle_received_frame(otInstance *instance,
 {
 	otRadioFrame recv_frame;
 
-	recv_frame.mPsdu = net_buf_frag_last(pkt->buffer)->data;
+	recv_frame.mPsdu = net_buf_frag_last(pkt->buffer) ?
+			   net_buf_frag_last(pkt->buffer)->data : NULL;
```

Comment:
And we should not call the function twice here as it causes the fragment list to be traversed twice.